existential-vs-universal-quantification/article.md

```diff
@@ -0,0 +1,295 @@
+# Existential and Universal Quantification in Haskell
+
+In this article I give a brief introduction into one of the most common Haskell extensions – `ExplicitForAll` – and syntax constructions that this extension brings to language.
+It is used in about all big haskell projects, so it is very nice to either become acquainted with it or to refresh it in your memory if you already know about it.
+
+Stay tuned if you want to know how to create lists of differently typed variables, how to add complex constraints, and more!
+
+## Introduction
+
+By default, quantification in Haskell is implicit.
+Without the `ExplicitForAll` extension, you can't write quantifiers. Moreover, all variables are universally quantified, and existential quantification is off.
+The key point of both universal and existential quantification is choosing where, we are going to instantiate a type variable - where it is defined, or where it is used.
```

Comment:
As you can see, the notion of instantiation can be confusing too. I would spare a couple of words to explain that instantiation is turning a variable into a concrete type. It's rather different, by the way, from what Java people think instantiation is, which -- in their case -- is all about bringing variables to scope.

https://stackoverflow.com/questions/23706331/terminology-what-does-it-mean-to-instantiate-type-variables-in-prof-giesls

Reply:
Done here: https://github.com/serokell/blog-posts/pull/4/commits/63b2e18f537a32f858d243ace6ccb563710a4095